### PR TITLE
Enable safe logging preconditions

### PR DESCRIPTION
**Pull request by schlosna**
Enables `PreferSafeLoggingPreconditions:ERROR` check and converts from Guava to log-safe preconditions checks

### CODE REVIEW

File: tritium-slf4j/build.gradle [hunk 1]
```diff
@@ -6,6 +6,7 @@ dependencies {
     api project(':tritium-core')
 
     implementation 'com.google.guava:guava'
+    implementation 'com.palantir.safe-logging:preconditions'
     implementation 'com.palantir.safe-logging:safe-logging'
```
Comment (carterkozak) [inline tritium-slf4j/build.gradle:10]: nit: can remove this line

Comment (carterkozak) [reply tritium-slf4j/build.gradle:10]: nvm, probably safest not to since we reference classes from both directly

**Review by carterkozak — APPROVED** [review]
lgtm. We need to be careful about using preconditions with SafeArg/UnsafeArg parameters on hot paths where we're better off using a conditional and throwing the exception ourselves. I don't see anything matching that pattern, and the error-prone check will only fail for uses that can be migrated without changing the exception message.

File: tritium-slf4j/src/main/java/com/palantir/tritium/event/log/LoggingInvocationEventHandler.java [hunk 2]
```diff
@@ -192,9 +191,6 @@ static String generateMessagePattern(int argCount) {
             message.append("{}");
         }
         message.append(") took {}ms");
-        Preconditions.checkState(estimatedSize == message.length(),
-                "Incorrect estimated size for %s arguments, expected %s, was %s",
-                argCount, estimatedSize, message.length());
```
Comment (schlosna) [inline tritium-slf4j/src/main/java/com/palantir/tritium/event/log/LoggingInvocationEventHandler.java:197]: this was the one potential hot path and this check was more of a perf optimization sanity check that is not needed

**Review by schlosna — COMMENTED** [review]
yeah, I checked all the places explicitly creating `Arg` and they aren't on hot paths.

**Pull request merged**